components/service/glean/src/main/java/mozilla/components/service/glean/storages/EventsStorageEngine.kt

```diff
@@ -133,11 +157,36 @@ internal object EventsStorageEngine : StorageEngine {
      */
     @Synchronized
     fun getSnapshot(storeName: String, clearStore: Boolean): List<RecordedEventData>? {
+        val file = getFile(storeName)
```

Comment:
Mh, are we hitting I/O every time we take a snapshot? What if we loaded the content of the file at startup, put that in a different map, e.g. `persistedEventsMap`, and if the map was non empty use it here?